Use Azure to build for macOS

jakirkham opened the pull request:
Workaround the backlog on Travis CI by switching macOS builds to Azure. Ensure that Azure is able to upload the packages as well.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

jakirkham committed:
Enable Azure to build macOS and Linux packages

It seems CircleCI and Travis CI have really long queues. So configure
Azure to build and upload packages for Linux and macOS to move things
along.

jakirkham committed:
MNT: Re-rendered with conda-build 3.17.5, conda-smithy 3.2.14, and conda-forge-pinning 2019.02.24

Now that Azure is configured to build and upload Linux and macOS
packages, re-render the feedstock to update the CI configuration files
and other content accordingly.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Revert Linux to CircleCI

As the CircleCI build on `master` just completed, revert Linux builds to
use CircleCI for now.

jakirkham committed:
MNT: Re-rendered with conda-build 3.17.5, conda-smithy 3.2.14, and conda-forge-pinning 2019.02.24

Now that Linux builds are configured to run on CircleCI again, re-render
to reconfigure the CIs accordingly.
